natchez-log4cats/src/main/scala/com/ovoenergy/effect/TracedLogger.scala

```diff
@@ -0,0 +1,90 @@
+package com.ovoenergy.effect
+
+import cats.Monad
+import cats.data.Kleisli
+import cats.effect.Sync
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import io.chrisdavenport.log4cats.StructuredLogger
+import natchez.{Kernel, Span, Trace}
+
+object TracedLogger {
+
+  private def lowercaseHeaders(kernel: Kernel): Map[String, String] =
+    kernel.toHeaders.map { case (k, v) => k.toLowerCase -> v }
+
+  /**
+   * Kernel to MDC for Datadog
+   * See docs at https://docs.datadoghq.com/logs/log_collection/java/?tab=log4j
+   */
+  private def datadogMdc(kernel: Kernel): Map[String, String] = {
+    val headers = lowercaseHeaders(kernel)
+    (
+      headers.get("x-parent-id").map("dd.span_id" -> _) ++
+      headers.get("x-trace-id").map("dd.trace_id" -> _)
+    ).toMap
```

Comment:
I haven't checked the implementation yet, but I remember you were saying something that at the moment you can't add arbitrary tags into the kernel, but there is an open PR in natchez that allows you to do that? Did I get this right?

Reply:
Yes it is this: https://github.com/tpolecat/natchez/pull/18

For our specific needs I think the trace ID & span ID is enough for now as then Datadog is able to correlate logs to traces, and the traces then have all the other tags

Reply:
that makes sense 👍 
I'm a little bit surprised because I thought that the "local" propagation was the default behaviour for any tags that you added to your parent span, perhaps I misunderstood how it worked.

Reply:
the `natchez-datadog` module in this library does indeed do local propagation currently though I am wondering about changing it at some point so it is more consistent with the other Natchez backends - Since all the spans are shown together in a trace I think propagating tags locally probably isn't actually that useful nowadays

Reply:
So when the logs are correlated with a span, does it mean that you can then filter/search them by the tags which are coming from the span?

Reply:
I don't think you can search logs with APM tags, unless you use `natchez-slf4j` which relays all the tracing information verbatim to SLF4J (while this module just correlates log4cats log lines with traces)

I think as we use Datadog more hopefully it'll become more obvious what we should do to make things as searchable as possible, I'll probably hold off making any changes until then